lib/rules/number-max-precision/index.mjs

```diff
@@ -188,8 +189,9 @@ const rule = (primary, secondaryOptions) => {
 		 */
 		function precisionInsideFunction(functionName, currentPrecision) {
 			const precisionForFunction = insideFunctions.get(functionName);
+			const hasPrecision = typeof precisionForFunction !== 'undefined';
 
-			if (isNumber(precisionForFunction)) return precisionForFunction;
```

Comment:
If `get` returns it's either a number or undefined because we already validated the values.